MarsRover/Validator.cs

```diff
@@ -31,15 +31,42 @@ private static bool AreAllCharsValid(string[] lines)
         {
             foreach (var line in lines)
             {
-                var charsInElementAreValid = line.All(c => c == 'O' || c == 'N');
-                if (!charsInElementAreValid) return false;
+                var charsInLineAreValid = line.All(c => c == 'O' || c == 'N');
+                if (!charsInLineAreValid) return false;
             }
             return true;
         }
 
+        public static bool IsValidDirection(string input)

```

Comment:
an alternative way of checking for a valid direction:

`var validDirections = new [] {"N", "S", "W", "E"};
return validDirections.Contains(input, StringComparison.InvariantCultureIgnoreCase);
`